### PR TITLE
Add code static check with reviewdog

### LINKED ISSUE #262: Support code golangci-lint by reviewdog
## Describe the feature
Pull request automatically triggers static code inspection, and automatically review

### PR DESCRIPTION
**Pull request by lbbniu**
#### What type of PR is this?

Enhancement

#### What this PR does / why we need it:

Add code static check

#### Which issue(s) this PR fixes:

Fixes #262
Static check specification adjustments

### CODE REVIEW

File: pkg/autoscaling/estimator/oom.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ func (e *OOMResourceEstimator) GetResourceEstimation(evpa *autoscalingapi.Effect
 		} else {
 			oomBumpUpRatio, err := strconv.ParseFloat(bumpUpRatio, 64)
 			if err != nil {
-				return nil, fmt.Errorf("Parse bumpUpRatio failed: %v. ", err)
+				return nil, fmt.Errorf("parse bumpUpRatio failed: %v. ", err)
```
Comment (qmhu) [inline pkg/autoscaling/estimator/oom.go:50]: We already have golint in CI, what's the benefit to use reviewdog?

Or if we add some staticcheck config in Makefile?

WDYT @mfanjie

File: .github/workflows/reviewdog.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+jobs:
+  golangci-lint:
+    name: runner / golangci-lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+
+      - name: golangci-lint
+        uses: reviewdog/action-golangci-lint@v2
```
Comment (qmhu) [inline .github/workflows/reviewdog.yml:12]: Can we remove this now? Since we already exec golint in build flow

Comment (lbbniu) [reply .github/workflows/reviewdog.yml:12]: 建议留着，如果不需要自动cr就移除了

Comment (lbbniu) [reply .github/workflows/reviewdog.yml:12]: 可以对其他贡献者自动的代码自动review

**Comment by qmhu** [comment]
/LGTM
@lbbniu  Please squash the commits before merge.

**Pull request merged**